src/type/ipc.d.ts

```diff
@@ -292,6 +292,19 @@ type IpcIHData = {
     args: [splitTextWhenPaste: SplitTextWhenPasteType];
     return: void;
   };
+
+  GET_SETTING: {
+    args: [key: keyof import("@/type/preload").ElectronStoreType];
+    return: unknown;
+  };
+
+  SET_SETTING: {
+    args: [
+      key: keyof import("@/type/preload").ElectronStoreType,
+      newValue: unknown
+    ];
+    return: unknown;
+  };
```

Comment:
ここってunknownでも大丈夫なんでしたっけ。。
型にunknownを書くと、returnしたあとの値を使えなくなる気がするのですが、どうだったかなぁと。。

- https://github.com/VOICEVOX/voicevox/pull/855

でこのあたりの型に詳しそうな @k-chop さんにちょっとお伺いしたいかもです 🙇

Reply:
はい、セルフレビューになってしまいますが・・・

IPCメッセージ自体の型付けの形式的にgenericsが使えないので、一度unknownに落とす必要がありました。
SET_SETTINGの方を例に出しますが、ここの型定義によって
```
ipcRendererInvoke("SET_SETTING", key, newValue)
```
の返り値がunknownになります。なのでこれを使う実装の方でasを使い、型の辻褄を合わせている感じですね。
```
  setSetting: async (key, newValue) => {
    return (await ipcRendererInvoke(
      "SET_SETTING",
      key,
      newValue
    )) as typeof newValue;
  }
```
unknownからのasは何も検証せずどんな型としても扱えてしまう安全ではないものですが、
このメソッドは↓のような型付けが行われており、 `typeof newValue` でここで定義した型を引用しているので特に問題はないはずです。
```
  setSetting<Key extends keyof ElectronStoreType>(
    key: Key,
    newValue: ElectronStoreType[Key]
  ): Promise<ElectronStoreType[Key]>;
```
IPCメッセージ自体の型付けはunknownで諦めて、electronの外側に公開するメソッドの型の層でしっかり型付けしている、といった感じです。

こんな感じで推論されます。
![スクリーンショット 2022-07-15 215523](https://user-images.githubusercontent.com/241973/179227632-26a0b4e4-08c5-46f0-825c-d7d219ede3ef.png)
![スクリーンショット 2022-07-15 215613](https://user-images.githubusercontent.com/241973/179227657-645963e6-3273-499a-9ec6-681dbab1f782.png)

Reply:
なるほどです！！こういうテクニック？もあるんですね！！

もしかしたらunkownを使わなくても良いように実装し直したほうが良いかもなのですが、根本から変更する必要があった理想なので、とりあえずこの方針で良いと思いました！

本当はgenericsが良い旨をTODOコメントで書くとかどうでしょう。
こんな感じとか･･･！
```suggestion
  // TODO: genericsが使用できないため、unknownで型宣言して実装時に型を付ける
  GET_SETTING: {
    args: [key: keyof import("@/type/preload").ElectronStoreType];
    return: unknown;
  };

  SET_SETTING: {
    args: [
      key: keyof import("@/type/preload").ElectronStoreType,
      newValue: unknown
    ];
    return: unknown;
  };
```

Reply:
確かに妥協なので、TODOコメントはあった方がよさそうです。
suggestion取り込ませていただきました！ありがとうございます。
合わせてコンフリクト解消も行っておきました。